--- Crypt-SSLeay-0.72/SSLeay.xs.orig
+++ Crypt-SSLeay-0.72/SSLeay.xs
@@ -512,16 +512,24 @@
 char *
 get_notBeforeString(cert)
          X509* cert
          CODE:
-            RETVAL = (char *)X509_get_notBefore(cert)->data;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+            RETVAL = (char *)ASN1_STRING_get0_data(X509_get0_notBefore(cert));
+#else
+            RETVAL = (char *)X509_get_notBefore(cert)->data;
+#endif
          OUTPUT:
             RETVAL
 
 char *
 get_notAfterString(cert)
          X509* cert
          CODE:
-            RETVAL = (char *)X509_get_notAfter(cert)->data;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+            RETVAL = (char *)ASN1_STRING_get0_data(X509_get0_notAfter(cert));
+#else
+            RETVAL = (char *)X509_get_notAfter(cert)->data;
+#endif
          OUTPUT:
             RETVAL
 
 MODULE = Crypt::SSLeay      PACKAGE = Crypt::SSLeay::Version    PREFIX = VERSION_
