--- Crypt-SSLeay-0.72/SSLeay.xs	2026-07-29 16:02:51.634511664 +0200
+++ Crypt-SSLeay-0.72/SSLeay.xs	2026-07-29 16:02:51.639241354 +0200
@@ -513,7 +513,11 @@
 get_notBeforeString(cert)
          X509* cert
          CODE:
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+            RETVAL = (char *)ASN1_STRING_get0_data(X509_get0_notBefore(cert));
+#else
             RETVAL = (char *)X509_get_notBefore(cert)->data;
+#endif
          OUTPUT:
             RETVAL
 
@@ -521,7 +525,11 @@
 get_notAfterString(cert)
          X509* cert
          CODE:
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+            RETVAL = (char *)ASN1_STRING_get0_data(X509_get0_notAfter(cert));
+#else
             RETVAL = (char *)X509_get_notAfter(cert)->data;
+#endif
          OUTPUT:
             RETVAL
 
