--- Crypt-SSLeay-0.37/SSLeay.xs.cryptdef	Tue Jun 11 10:58:56 2002
+++ Crypt-SSLeay-0.37/SSLeay.xs	Tue Jun 11 10:59:24 2002
@@ -9,14 +9,14 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
-#include "EXTERN.h"
-#include "perl.h"
-#include "XSUB.h"
 
 /* build problem under openssl 0.9.6 and some builds of perl 5.8.x */
 #ifndef PERL5
 #define PERL5 1
 #endif
+#include "EXTERN.h"
+#include "perl.h"
+#include "XSUB.h"
 
 /* ssl.h or openssl/ssl.h is included from the crypt_ssleay_version
  * file which is written when building with perl Makefile.PL
